### PR TITLE
[Fix](function) Fix unix_timestamp core for string input

### PR DESCRIPTION
**Pull request by zclllyybb**
## Proposed changes

Issue Number: close #xxx

before when we got `unix_timestamp(string_col, literal)`, BE will core. now fixed.

**no need to pick to branch 2.0**

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by zclllyybb** [comment]
run buildall

File: regression-test/suites/query_p0/sql_functions/datetime_functions/test_date_function.groovy [hunk 1]
```diff
@@ -734,4 +734,17 @@ suite("test_date_function") {
...
+
+    sql """
+        create table date_varchar(
+            dt varchar null,
+            fmt varchar null
+        )
+        DISTRIBUTED BY HASH(`dt`) BUCKETS 1
+        properties("replication_num" = "1");
+    """
+    sql """ insert into date_varchar values ("2020-12-12", "%Y-%m-%d"), ("20201111", "%Y%m%d"), ("202012-13", "%Y%m-%d"); """
```
Comment (yiguolei) [inline regression-test/suites/query_p0/sql_functions/datetime_functions/test_date_function.groovy:746]: also test some invalid values like 0000-00-00, and some very large values 9999-12-31

Comment (yiguolei) [reply regression-test/suites/query_p0/sql_functions/datetime_functions/test_date_function.groovy:746]: we should define the valid range of input parameters and add some invalid parameter tests.

Comment (zclllyybb) [reply regression-test/suites/query_p0/sql_functions/datetime_functions/test_date_function.groovy]: added some. and more valid/invalid ranges has ensured in other origin tests

**Review by yiguolei — APPROVED** [review]
LGTM

**Review by HappenLee — APPROVED** [review]
LGTM

**Pull request merged**